### PR TITLE
Remove assignment col arg from BaseGraph

### PR DESCRIPTION
**Pull request by bsuwal**
This PR removes the requirement for BaseGraph to take an `assignment` argument. This issue has been brought up in the past -- "Why is BaseGraph and Partition both taking the assignment column, isn't it enough to pass it into just one?". In removing the `assignment` argument requirement from BaseGraph it makes the code more elegant. 

Previous usage:
```
graph = BaseGraph(filepath, POP_COL, ASSIGNMENT_COL)
partition = Partition(graph, ASSIGNMENT_COL)
```
New usage:
```
graph = BaseGraph(filepath, POP_COL)
partition = Partition(graph, ASSIGNMENT_COL)
```

This change was actually forced, in my opinion, by me trying to add the functionality of adding seed partition generating methods. In a non-trivial number of occasions at the lab we work with shapefiles without assignments and we generate a seed partition. If it is a requirement for the `BaseGraph` to take in an assignment column when we don't have one, we cannot even instantiate the `BaseGraph`! 

What this means:
* The `num_dists` property in `BaseGraph` gets transferred over to `Partition`, and this leads to a bunch of changes in the codebase.
* This paves way to new PRs on the pipeline: 1. removing the `population_col` argument in `PopulationConstraint` and 2. A PR to generate seed partitions!

That is the only major change this PR makes -- it touches a lot of files because the instantiation of `BaseGraph` is everywhere, especially in the tests.

Reminder to self / Promise: Need to update the documentation after the PR is merged!

### CODE REVIEW

File: src/graph.jl [hunk 1]
```diff
@@ -3,7 +3,6 @@ abstract type AbstractGraph end
 struct BaseGraph<:AbstractGraph
     num_nodes::Int
     num_edges::Int
-    num_dists::Int
```
Comment (bsuwal) [inline src/graph.jl:6]: Moving this property over to `Partition`

File: src/graph.jl [hunk 2]
```diff
@@ -70,41 +69,6 @@ function population_to_int(raw_value::Number)::Int
     return raw_value isa Int ? raw_value : convert(Int, round(raw_value))
 end
 
-
-function get_assignments(node_attributes::Array,
```
Comment (bsuwal) [inline src/graph.jl:74]: Moving this over to partition.jl since `BaseGraph` doesn't use it anymore.

File: src/partition.jl [hunk 3]
```diff
@@ -95,13 +130,12 @@ function get_district_adj_and_cut_edges(graph::BaseGraph,
 end
 
 function sample_adjacent_districts_randomly(partition::Partition,
-                                            num_dists::Int,
```
Comment (bsuwal) [inline src/partition.jl:98]: removing this argument because `num_dists` is embedded into `Partition` now.

**Review by pjrule — APPROVED** [review]
This will probably break some of our experiments that use `num_dists`, etc. 'Tis the price of progress, I suppose.

**Pull request merged**